custom/icds/models.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.app_manager.dbaccessors import get_build_by_version
+from corehq.motech.utils import b64_aes_decrypt
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+from custom.icds.validators import (
+    LowercaseAlphanumbericValidator,
+)
+
+
+class CCZHostingLink(models.Model):
+    identifier = models.CharField(null=False, unique=True, max_length=255, db_index=True,
+                                  validators=[LowercaseAlphanumbericValidator])
+    username = models.CharField(null=False, max_length=255)
+    # b64_aes_encrypt'ed raw password is stored in DB
+    password = models.CharField(null=False, max_length=255)
```

Comment:
While developing/testing i found it useful to see what the password is to a link. If we keep it one way we would need to keep storing passwords in keepasses again whereas this makes it possible to have things setup just on HQ. I am not sure if we consider this bad security because these are not user accounts so at some level this should be okay to be visible on HQ.
We have used the same implementation at more places in HQ so i just built the same.
Open to reconsideration if this is still concerning

Reply:
Updated to store passwords as salted hashed password.